packages/dds/tree/src/tree/types/index.ts

```diff
@@ -0,0 +1,8 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export type TreeId = number | string & { readonly TreeId: symbol; };
```

Comment:
Why is this `number | string`? Also, I'm not seeing this used anywhere else. Is that intentional?